posts/epi-community-contrib/index.qmd

```diff
@@ -0,0 +1,147 @@
+   isolated = \(n) rnbinom(n = n, mu = 0, size = 1),
+   asymptomatic = \(n) rnbinom(n = n, mu = 2.5, size = 0.16)
+  ),
+ delays = delay_opts(
+   incubation_period = \(n) rweibull(n = n, shape = 2.32, scale = 6.49),
+   onset_to_isolation = \(n) rweibull(n = n, shape = 2.5, scale = 5)
+  ),
+ event_probs = event_prob_opts(
+   asymptomatic = 0,
+   presymptomatic_transmission = 0.3,
+   symptomatic_ascertained = 0
+  ),
+ interventions = intervention_opts(quarantine = TRUE),
+ sim = sim_opts(
+   cap_max_days = 365,
+   cap_cases = 2000
+  )
+)
+```
+
+:::
+
+::::
+
+The new API gives the user more control over the model's parameterisation. The incubation period is now specified by the user instead of being set to an estimate for COVID-19. The way offspring and delay distribution functions are specified also means that any distributional or non-parametric form can be supplied, relaxing the assumption that the onset-to-isolation has to be a Weibull distribution.
+
+Users can now specify the proportion of presymptomatic transmission rather than having to understand the skew normal parameterisation used by the simulation model, making it easier to get started with the package for new users.
+
+Lastly on user-facing changes, the naming and style of function arguments has been standardised for consistent use of [snakecase](https://en.wikipedia.org/wiki/Snake_case) style and abbreviations.  
+
+### Documentation
+
+Function documentation already used {roxygen2}, but did not make use of inheritance or comprehensively document the function output or usage. We used `@inheritParams` from {roxygen2} to deduplicate, added `@return` documentation to all functions. We also improved the function argument documentation by following a structure of: `<type>: description`, for example:
+
+```r
+@param sim a `list` with class `<ringbp_sim_opts>`: the simulation control
+  options for the \pkg{ringbp} model, returned by [sim_opts()]
+```
+
+Exported functions now have informative examples (`@examples`) to showcase how the functions should be used. Function examples now always run (removing `\dontrun{}`) to catch any errors.
+
+The [{roxyglobals}](https://github.com/anthonynorth/roxyglobals) package has been added to automate the management of global variables with the use of the `@autoglobal` tag.
+
+Vignettes are useful long-form package documentation. Thus far we've added one vignette to the package and plan to add more where relevant.
+
+### Bug fixes
+
+Perhaps more important that any of the software best practices and user interface is the correctness of the code. In our developments we've uncovered a few bugs in the previous version of {ringbp}. Errors in the timing of quarantining infected individuals, sampling from the onset-to-isolation distribution, and calculating the generation time from the incubation period have all been identified and fixed.
+
+### Testing
+
+- simulation correctness regression (snapshot) testing
```

Comment:
Should this be a sentence so that it follows the rest of the write up?

Reply:
Yes this was a placeholder, I've now added a short paragraph on snapshot testing here.